src/test/java/api/loans/ChangeDueDateAPITests.java

```diff
@@ -0,0 +1,285 @@
+
+    changeDueDateFixture.changeDueDate(new ChangeDueDateRequestBuilder()
+      .forLoan(loan.getId())
+      .withDueDate(newDueDate));
+
+    Response response = loansClient.getById(loan.getId());
+
+    JsonObject updatedLoan = response.getJson();
+
+    assertThat("due date is not updated",
+      updatedLoan.getString("dueDate"), isEquivalentTo(newDueDate));
+  }
+
+  @Test
+  public void cannotChangeDueDateWhenDueDateIsNotProvided() {
+    final Response response = changeDueDateFixture
+      .attemptChangeDueDate(new ChangeDueDateRequestBuilder()
+        .forLoan(loan.getId())
+        .withDueDate(null));
+
+    assertResponseOf(response, 422, DUE_DATE);
+    assertResponseMessage(response, "Due date is a required field");
+  }
+
+  @Test
+  public void cannotChangeDueDateWhenLoanIsNotFound() {
+    final String nonExistentLoanId = UUID.randomUUID().toString();
+    final DateTime newDueDate = dueDate.plus(Period.days(14));
+
+    final Response response = changeDueDateFixture
+      .attemptChangeDueDate(new ChangeDueDateRequestBuilder()
+        .forLoan(nonExistentLoanId)
+        .withDueDate(newDueDate));
+
+    assertThat(response, hasStatus(HTTP_NOT_FOUND));
+  }
+
+  @Test
+  public void cannotChangeDueDateWhenClosed() {
+    final DateTime newDueDate = dueDate.plus(Period.days(14));
+
+    checkInFixture.checkInByBarcode(item);
+
+    final Response response = changeDueDateFixture
+      .attemptChangeDueDate(new ChangeDueDateRequestBuilder()
+        .forLoan(loan.getId())
+        .withDueDate(newDueDate));
+
+    assertResponseOf(response, 422, "id", loan.getId());
+    assertResponseMessage(response, "Loan is closed");
+  }
+
+  @Test
+  public void cannotChangeDueDateWhenDeclaredLost() {
+    final DateTime newDueDate = dueDate.plus(Period.days(14));
+
+    assertThat(loansFixture.declareItemLost(loan.getJson()),
+      hasStatus(HTTP_NO_CONTENT));
+
+    final Response response = changeDueDateFixture
+      .attemptChangeDueDate(new ChangeDueDateRequestBuilder()
+        .forLoan(loan.getId())
+        .withDueDate(newDueDate));
+
+    assertResponseOf(response, 422, ITEM_ID, item.getId());
+    assertResponseMessage(response, "item is Declared lost");
+  }
+
+  @Test
+  public void cannotChangeDueDateWhenClaimedReturned() {
+    final DateTime newDueDate = dueDate.plus(Period.days(14));
+
+    assertThat(loansFixture.claimItemReturned(
+      new ClaimItemReturnedRequestBuilder()
+        .forLoan(loan.getId().toString())
+       ), hasStatus(HTTP_NO_CONTENT));
+
+    (new ChangeDueDateRequestBuilder()).forLoan(loan.getId().toString());
+
+    final Response response = changeDueDateFixture
+      .attemptChangeDueDate(new ChangeDueDateRequestBuilder()
+        .forLoan(loan.getId())
+        .withDueDate(newDueDate));
+
+    assertResponseOf(response, 422, ITEM_ID, item.getId());
+    assertResponseMessage(response, "item is Claimed returned");
+  }
+
+  @Test
+  public void cannotReapplyDueDateWhenClaimedReturned() {
```

Comment:
What does re-apply mean? Does that mean a change due date cannot be done more than once?

Reply:
The purpose of this was to test that the date is being changed with the same value.
In the older design, when updating the entire Loan instead of just the Due Date, assigning the same value has certain implications and is prohibited (if my memory serves me correctly).

This test is done to assure that the due date can be updated even if it is the same exact date.
The word _reapply_ was my best attempt to find a word to describe this.

Reply:
@kaladay 

> This test is done to assure that the due date can be updated even if it is the same exact date.

Do you mean cannot? The test name suggests it is cannot.

Reply:
Yes, sorry. I must have swapped my can and cannot in my comment.

The _due date_ is being prohibited from being _re-applied_ when the item is _claimed returned_.
This test is to confirm/deny that behavior.

Reply:
Thanks @kaladay  

In which case, I suggest we name it something like:

```suggestion
  public void cannotChangeDueDateToSameDateWhenItemIsClaimedReturned() {
```

What do you think?

Reply:
I have no complaints with this rename.